grafana-plugin/src/components/Timeline/TimelineItem.tsx

```diff
@@ -22,14 +22,17 @@ const TimelineItem: React.FC<TimelineItemProps> = ({
   contentClassName,
   children,
   isDisabled,
-  backgroundColor = '#3274D9',
+  backgroundColor = { color: '#3274D9', isClassName: false },
   textColor = '#ffffff',
   number,
 }) => {
   return (
     <li className={cx('item', className)}>
       {!isDisabled && (
-        <div className={cx('dot')} style={{ backgroundColor, color: textColor }}>
+        <div
+          className={cx('dot', backgroundColor.isClassName ? backgroundColor.color : '')}
+          style={{ backgroundColor: backgroundColor.isClassName ? '' : backgroundColor.color, color: textColor }}
```

Comment:
1. Do we need to apply color as style? Can't we just rely on class?
2. If we really want to keep both, then what do you think about having 2 separate props: `numberClassName: string` and `numberHexColor: string`? `isClassName` is a bit confusing imho and also `backgroundColor.color` doesn't necessarily need to be a class that applies color, it's just any class that is attached to the div

wdyt?

Reply:
This was actually a bugfix and not a needed implementation change. We need both because one is read from emotion and the other is passed as a hex number. I can rename those

Reply:
What about using 2 separate props instead of `backgroundColor` object with `isClassName` boolean as mentioned in point 2 above?